cmd/fs-v1-multipart.go

```diff
@@ -474,7 +605,22 @@ func (fs fsObjects) CompleteMultipartUpload(bucket string, object string, upload
 		return "", err
 	}
 
-	uploadIDPath := path.Join(bucket, object, uploadID)
+	bucketDir, err := fs.getBucketDir(bucket)
+	if err != nil {
+		return "", toObjectErr(err, bucket)
+	}
+
+	if _, err = fsStatDir(bucketDir); err != nil {
+		return "", toObjectErr(traceError(err), bucket)
+	}
+
+	uploadIDPath := pathJoin(bucket, object, uploadID)
+
+	rwlk, err := fs.rwPool.Create(pathJoin(fs.fsPath, minioMetaMultipartBucket, bucket, object, uploadsJSONFile))
+	if err != nil {
+		return "", toObjectErr(traceError(err), bucket, object)
+	}
+	defer rwlk.Close()
```

Comment:
sequence of lock should always be NSLock and then fcntl lock, else fcntl locks will merge.

for ex.
* fs.rwPool.Open at some other place succeeds
* fs.rwPool.Create succeeds (merge happens) i.e read and write locks are "held" at the same time.
* fd closed got using fs.rwPool.Open closes, this will release the lock held by fs.rwPool.Create which will cause problems.